packages/cadl-ranch-specs/http/type/model/templated/main.tsp

```diff
@@ -0,0 +1,72 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+/**
+ * Illustrates the model templated cases. There is a base templated type and an instantiated type extending from it.
+ */
+@scenarioService("/type/model/templated")
+namespace Type.Model.Templated;
+
+@friendlyName("{name}Type", T)
+model NumericType<T extends numeric> {
+  /**
+   * An array of numeric values.
+   */
+  values: T[];
+
+  value: T;
+}
+
+/**
+ * An instantiated type representing int32 type.
+ */
+model Int32Type extends NumericType<int32> {
```

Comment:
so this is one scenario that we extends a generic type.
I think we need to add some more operation takes `NumericType<T>` directly.